frontend/appflowy_flutter/lib/plugins/database/widgets/cell/desktop_row_detail/desktop_row_detail_url_cell.dart

```diff
@@ -18,13 +19,86 @@ class DesktopRowDetailURLSkin extends IEditableURLCellSkin {
     TextEditingController textEditingController,
     URLCellDataNotifier cellDataNotifier,
   ) {
-    return TextField(
+    return LinkTextField(
       controller: textEditingController,
       focusNode: focusNode,
+    );
+  }
+
+  @override
+  List<GridCellAccessoryBuilder> accessoryBuilder(
+    GridCellAccessoryBuildContext context,
+    URLCellDataNotifier cellDataNotifier,
+  ) {
+    return [
+      accessoryFromType(
+        GridURLCellAccessoryType.visitURL,
+        cellDataNotifier,
+      ),
+    ];
+  }
+}
+
+class LinkTextField extends StatefulWidget {
+  const LinkTextField({
+    super.key,
+    required this.controller,
+    required this.focusNode,
+  });
+
+  final TextEditingController controller;
+  final FocusNode focusNode;
+  @override
+  State<LinkTextField> createState() => _LinkTextFieldState();
+}
+
+class _LinkTextFieldState extends State<LinkTextField> {
+  bool _isLinkClickable = false;
+
+  @override
+  void initState() {
+    super.initState();
+    HardwareKeyboard.instance.addHandler(_handleGlobalKeyEvent);
+  }
+
+  @override
+  void dispose() {
+    HardwareKeyboard.instance.removeHandler(_handleGlobalKeyEvent);
+    super.dispose();
+  }
+
+  bool _handleGlobalKeyEvent(KeyEvent event) {
+    setState(() {
+      _isLinkClickable = event is KeyDownEvent &&
+          [
+            LogicalKeyboardKey.control,
+            LogicalKeyboardKey.controlLeft,
+            LogicalKeyboardKey.controlRight,
+            LogicalKeyboardKey.meta,
+            LogicalKeyboardKey.metaLeft,
+            LogicalKeyboardKey.metaRight,
+          ].contains(event.logicalKey);
+    });
+
+    return false;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return TextField(
+      mouseCursor:
+          _isLinkClickable ? SystemMouseCursors.click : SystemMouseCursors.text,
+      controller: widget.controller,
+      focusNode: widget.focusNode,
       style: Theme.of(context).textTheme.bodyMedium?.copyWith(
             color: Theme.of(context).colorScheme.primary,
             decoration: TextDecoration.underline,
           ),
+      onTap: () {
+        if (_isLinkClickable) {
+          openUrlCellLink(widget.controller.text);
+        }
+      },
```

Comment:
Couldn't this widget be stateless if you did it like this:
```suggestion
      onTap: () {
        final keyboard = HardwareKeyboard.instance;
        if (keyboard.isControlPressed || keyboard.isMetaPressed) {
          openUrlCellLink(widget.controller.text);
        }
      },
```

Reply:
This is true for `onTap`, but there is still the `mouseCursor` state to consider. Do you recommend we scrap the cursor visual change to make this widget stateless?

https://github.com/AppFlowy-IO/AppFlowy/blob/dbf3c4a0ec4b51ed13d7de7c931b5773ea514951/frontend/appflowy_flutter/lib/plugins/database/widgets/cell/desktop_row_detail/desktop_row_detail_url_cell.dart#L89-L90

Reply:
I see, let's keep it.

I've made another comment for minor improvement instead if we keep the state. 👍